src/lib/components/switch/QSwitch.scss

```diff
@@ -0,0 +1,268 @@
+@use "sass:list";
+@use "$css/mixins";
+
+$root: ".q-switch";
+$track: "#{$root}__track";
+$handle: "#{$root}__handle";
+$icon: "#{$root}__icon";
+
+$checked: "[aria-checked]";
+$unchecked: ":not([aria-checked])";
+
+$check-map: (
+  "checked": "[aria-checked]",
+  "unchecked": ":not([aria-checked])",
+);
+
+@function getColor($el, $checked: false, $hovered: null) {
+  @if $el == "track" {
+    @return if($checked, var(--primary), var(--surface-container-highest));
+  } @else if $el == "icon" {
+    @return if($checked, var(--on-primary-container), var(--surface-container-highest));
+  } @else if $el == "handle" {
+    @if ($checked) {
+      @return if($hovered, var(--primary-container), var(--on-primary));
+    } @else {
+      @return if($hovered, var(--on-surface-variant), var(--outline));
+    }
+  }
+
+  @error "The given element #{$el} is not supported.";
+}
+
+.q-switch {
+  position: relative;
+
+  display: inline-flex;
+  align-items: center;
+  flex-shrink: 0; // Stop from collapsing in flex containers
+  gap: 0.8rem;
+
+  cursor: pointer;
+
+  outline: none;
+
+  &--reversed {
+    flex-direction: row-reverse;
+  }
+
+  &__inner {
+    width: 3.25rem;
+    height: 2rem;
+  }
+
+  &[aria-disabled] :global(:is(.q-switch__track, .q-switch__handle)) {
+    @include mixins.disabled;
+  }
+
+  & :global(input) {
+    appearance: none;
+    outline: none;
+    margin: 0;
+    position: absolute;
+    width: 100%;
+    z-index: 1;
+    cursor: inherit;
+  }
+}
+
+// Track
+.q-switch {
+  & :global(#{$track}) {
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    box-sizing: border-box;
+
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    border-radius: 10rem;
+    cursor: pointer;
+
+    &::before {
+      content: "";
+      position: absolute;
+      height: 100%;
+      width: 100%;
+
+      display: flex;
+
+      border-radius: inherit;
+      box-sizing: border-box;
+      transition-property: opacity, background-color;
+      transition-timing-function: linear;
+      transition-duration: 67ms;
+    }
+  }
+
+  @each $state, $selector in $check-map {
+    &#{$selector} :global(#{$track}::before) {
+      background-color: getColor("track", $state == "checked");
+    }
+  }
+
+  &#{$checked} :global(#{$track}) {
+    background-color: transparent;
+    border-color: transparent;
+
+    &::before {
+      background-clip: content-box;
+    }
+  }
+
+  &#{$unchecked} :global(#{$track}::before) {
+    border-color: var(--outline);
+    border-style: solid;
+    border-width: 0.125rem;
+  }
+
+  &[aria-disabled] :global(#{$track}:is(::before, ::after)) {
+    transition: none;
+    opacity: 38%;
+  }
+}
+
+// Handle container
+.q-switch {
+  & :global(#{$handle}-container) {
+    position: relative;
+
+    display: flex;
+    place-content: center;
+    place-items: center;
+
+    // cubic-bezier(0.13, 0.78, 0.33, 1.15) for little bounce
```

Comment:
?

Reply:
It's for later, if we implement the possibility to use bouncy animations